package.json

```
@@ -7,6 +7,7 @@
     "test": "test"
   },
   "scripts": {
+    "lint" : "eslint ./test/**/*.js",
     "test": "mocha -t 5000 --reporter mochawesome --reporter-options reportDir=report,reportFilename=ApiTesting"
```

Debes bajar el lint para donde estaba.  Osea debajo de la linea de test, con esto evitas el conflicto.
Con respecto a la sugerencia anterior de ejecutar primero el lint antes de mocha dentro del comando test, es basicamente hacer lo siguiente:
```suggestion
    "test": "lint && mocha -t 5000 --reporter mochawesome --reporter-options reportDir=report,reportFilename=ApiTesting",
```

La razon principal de este cambio es que el linter ejecuta mas rapido que los tests, ademas, que el linter te ayuda a encontrar posibles errores de sintaxis o problemas de codigo limpio.